### PR TITLE
[DOC] Adjustments to links in array.c

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Mostly suppressing links to itself.

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -7849,15 +7849,15 @@ rb_ary_deconstruct(VALUE ary)
  *
  *  You can convert certain objects to Arrays with:
  *
- *  - \Method #Array.
+ *  - \Method {Array}[Kernel.html#method-i-Array].
```
Comment (zverok) [inline array.c:7852]: What's the reason of this change? Even if changing references to explicit, we should use `rdoc-ref:` scheme for such cases (e.g. `[rdoc-ref:Kernel#Array]`), but I think `#Array` [works fine currently](https://docs.ruby-lang.org/en/3.1/Array.html#class-Array-label-Creating+Arrays)?..

**Review by zverok — COMMENTED** [review]
Other than one puzzling link change, I don't see problems here :+1:

File: array.c (hunk 1 above)
Comment (BurdetteLamar) [reply array.c]: Reverted.

**Review by zverok — APPROVED** [review]
Thanks!
